Directory.Build.props

```diff
@@ -0,0 +1,25 @@
+<Project>
+
+  <!-- Contains global properties for all projects in the solution.
+  These can be overridden in inner folders, if necessary. 
+  See more here https://docs.microsoft.com/en-us/visualstudio/msbuild/customize-your-build?view=vs-2019 -->
+  
+  <PropertyGroup>
+    <TargetIsNetFx Condition="$(TargetFramework.StartsWith('net4'))">true</TargetIsNetFx>
+    <LangVersion>latest</LangVersion>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+    <WarningsAsErrors />
+  </PropertyGroup>
```

Comment:
Is it indeed common? Didn't we allow to have warning in test files before?

Reply:
Yes, we did warnings in tests before. I wanted to ask if that was intentional or not. If it was, I can either override it inside the `test` folder with another `props` file, or just move these two properties back to the main `NSubstitute` project.

Reply:
I would do it like that for now to not solve even more issues within this PR. It should also solve the CI issue you faced with dynamically generated project.

Later we could always improve that via a separate PR.

Reply:
> It should also solve the CI issue you faced with dynamically generated project.

Maybe can just override this for the dynamically generated project?

https://github.com/nsubstitute/NSubstitute/blob/d02bbb96836111d8a90436ca8802c98a73eb04f0/build/build.fsx#L166

Reply:
Meaning you really want to enforce this? Well, I’m fine by that as well, was just looking for the simpler way 🤝

Reply:
Not being lazy but.. I guess it makes sense to think about it better and change it in a later PR.